### ISSUE TITLE
Redis attributes not aligned with latest semantic conventions

### ISSUE THREAD

**Issue by luke6Lh43** [open-telemetry/opentelemetry-python-contrib#2885]
### Describe your environment

OS: macOS Sequoia
Architecture: x86_64 (Intel processor)
Python: 3.12.5

opentelemetry-api==1.27.0
opentelemetry-distro==0.48b0
opentelemetry-exporter-otlp==1.27.0
opentelemetry-exporter-otlp-proto-common==1.27.0
opentelemetry-exporter-otlp-proto-grpc==1.27.0
opentelemetry-exporter-otlp-proto-http==1.27.0
opentelemetry-instrumentation==0.48b0
opentelemetry-instrumentation-redis==0.48b0
opentelemetry-proto==1.27.0
opentelemetry-sdk==1.27.0
opentelemetry-semantic-conventions==0.48b0

### What happened?

Hello,

I have recently been working on a project that required Redis instrumentation for two different languages. During this process, I noticed some discrepancies in the attributes being collected. To investigate further, I conducted testing to check the attributes collected for Redis instrumentation across six of the most popular languages. I have documented my testing in this GitHub repository:

[[Link to GitHub Repo]](https://github.com/luke6Lh43/otel-redis-testing)

What We Capture Today ==> How It Should Look in Latest Semantic Conventions

db.statement: GET ? ==> db.statement: GET ?
db.system: redis ==> db.system: redis
db.redis.database_index: 0 ==> db.namespace: 0
net.peer.name: localhost ==> server.address: localhost
net.peer.port: 6379 ==> network.peer.port: 6379
net.transport: ip_tcp ==> Not sure if it's needed
db.redis.args_length: 2 ==> Not sure if it's needed

The latest OpenTelemetry semantic conventions for Redis can be found here:
[OpenTelemetry Semantic Conventions for Redis](https://opentelemetry.io/docs/specs/semconv/database/redis/)

### Steps to Reproduce

https://github.com/luke6Lh43/otel-redis-testing/blob/main/Readme.md#python

### Expected Result

See above

### Actual Result

See above

### Would you like to implement a fix?

None

**Comment by luke6Lh43** [comment]
### Describe your environment

OS: macOS Sequoia
Architecture: x86_64 (Intel processor)
Python: 3.13.3

opentelemetry-api==1.37.0
opentelemetry-distro==0.58b0
opentelemetry-exporter-otlp==1.23.0
opentelemetry-exporter-otlp-proto-common==1.37.0
opentelemetry-exporter-otlp-proto-grpc==1.37.0
opentelemetry-exporter-otlp-proto-http==1.37.0
opentelemetry-instrumentation==0.58b0
opentelemetry-instrumentation-redis==0.58b0
opentelemetry-proto==1.37.0
opentelemetry-sdk==1.37.0
opentelemetry-semantic-conventions==0.58b0

### What happened?

Hi team,

This is a follow-up to my previous report regarding the OpenTelemetry Python Redis instrumentation's adherence to the latest semantic conventions.

I've just re-tested the instrumentation with the latest available versions and observed that some discrepancies still persist between the generated spans and the current Redis semantic conventions.

**Observed Span (for a simple GET operation):**

```console
{
    "name": "GET",
    "context": {
        "trace_id": "0xc4d7fa5faf473092230376d15ad15f51",
        "span_id": "0xf520caf13ccc84cc",
        "trace_state": "[]"
    },
    "kind": "SpanKind.CLIENT",
    "parent_id": null,
    "start_time": "2025-09-15T00:23:00.574719Z",
    "end_time": "2025-09-15T00:23:00.575428Z",
    "status": {
        "status_code": "UNSET"
    },
    "attributes": {
        "db.statement": "GET ?",
        "db.system": "redis",
        "db.redis.database_index": 0,
        "net.peer.name": "localhost",
        "net.peer.port": 6379,
        "net.transport": "ip_tcp",
        "db.redis.args_length": 2
    },
    "events": [],
    "links": [],
    "resource": {
        "attributes": {
            "telemetry.sdk.language": "python",
            "telemetry.sdk.name": "opentelemetry",
            "telemetry.sdk.version": "1.37.0",
            "service.name": "python-redis-testing",
            "telemetry.auto.version": "0.58b0"
        },
        "schema_url": ""
    }
}
```

### Discrepancies from Current Redis Semantic Conventions:

Based on the [Redis Semantic Conventions (Status: Development)](https://opentelemetry.io/docs/specs/semconv/database/redis/), I've identified the following points for improvement:

**Incorrect Attribute Name for Database System:**

The span uses "db.system": "redis". The convention specifies db.system.name as the attribute for the database system.
Expected: "db.system.name": "redis"

**Missing db.operation.name (Required):**

The convention states db.operation.name is a Required attribute for the Redis command name. While the span name is "GET", this attribute itself is missing from attributes.
Expected: "db.operation.name": "GET"

**Incorrect Attribute Name and Type for Database Index:**

The span uses "db.redis.database_index": 0. The convention specifies db.namespace and explicitly states it should be a string (e.g., "0", "1").
Expected: "db.namespace": "0"

**Use of Deprecated Network Attributes:**

The span uses "net.peer.name": "localhost" and "net.peer.port": 6379. The latest conventions recommend server.address and server.port for the database host and port.
Expected: "server.address": "localhost" and "server.port": 6379 (along with optional network.peer.address and network.peer.port if distinguishing network peer from server).

**Incorrect Attribute Name for Query Text:**

The span uses "db.statement": "GET ?". The convention recommends db.query.text for the full syntax of the Redis CLI command.
Expected: "db.query.text": "GET ?"

**Extraneous Attribute:**

"db.redis.args_length": 2 is present but is not part of the current Redis semantic conventions. This attribute should ideally be removed.

### Questions

* Is the OpenTelemetry Python community aware of these specific discrepancies with the latest semantic conventions?
* Are there any plans to update the instrumentation to fully align with the current Redis Semantic Conventions?
* Given that the "Semantic Conventions for Redis" document is still in "Development" status, should we expect these changes to be implemented only once the document reaches a stable state, or are updates planned incrementally?

**Comment by tammy-baylis-swi** [comment]
Closing this as completed via https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2930
